### ISSUE TITLE
filter CC and NRPN data from receiving and or logging

### ISSUE THREAD

**Issue by markusschloesser** [christofmuc/KnobKraft-orm#241]
My Andromeda often sends continues CC or NRPN data (common problem with sticky knobs on that synth), and KK always goes crazy in terms of CPU. This also happens when I turn encoders or automate from the DAW and have KK open. The log file quickly goes into the 10.000th. 
Wouldn't it make sense to filter/not receive CC/NRPN data? Or if you need it for certain things (program change), filter for exactly that PC data?

**Comment by christofmuc** [comment]
Ah, good point. I wanted to add filters to the MIDI log like in MIDI Ox, and then got unhappy with the way the filters work for the error log (namely, they don't). This is a bit because I am misusing the document editor window as the log window, and it might slow down with a very large document... 

I'll think about adding filter buttons!

**Comment by markusschloesser** [comment]
If I have kk open in the background and am constantly tweaking /turning knobs, it also becomes slow after a while, most probably because of the ten thousand log entries

**Comment by christofmuc** [comment]
Ah, this is part of 2.0.3. Forgot to add it to the release notes. By default, only sysex messages are logged now, and if you want to see all you can use the MIDI menu to switch to full logging.

**Comment by markusschloesser** [comment]
1. doesn't seem to work here, Midi log shows (for example)
```05:30:46.721: Out 2- DIN8 MKS70 Channel pressure 57 Channel 1 [d0 39]
05:30:46.726: Out 2- DIN8 MKS70 Controller Sound Brightness: 42 Channel 1 [b0 4a 2a]
05:30:46.728: Out 2- DIN8 MKS70 Channel pressure 55 Channel 1 [d0 37]
05:30:46.735: Out 2- DIN8 MKS70 Channel pressure 53 Channel 1 [d0 35]
05:30:46.742: Out 2- DIN8 MKS70 Channel pressure 51 Channel 1 [d0 33]
05:30:46.749: Out 2- DIN8 MKS70 Channel pressure 49 Channel 1 [d0 31]
05:30:46.756: Out 2- DIN8 MKS70 Channel pressure 47 Channel 1 [d0 2f]
05:30:46.763: Out 2- DIN8 MKS70 Channel pressure 45 Channel 1 [d0 2d]
05:30:46.770: Out 2- DIN8 MKS70 Channel pressure 43 Channel 1 [d0 2b]
05:30:46.777: Out 2- DIN8 MKS70 Channel pressure 41 Channel 1 [d0 29]
05:30:46.784: Out 2- DIN8 MKS70 Channel pressure 39 Channel 1 [d0 27]
05:30:46.791: Out 2- DIN8 MKS70 Channel pressure 37 Channel 1 [d0 25]
05:30:46.798: Out 2- DIN8 MKS70 Channel pressure 35 Channel 1 [d0 23]
05:30:46.805: Out 2- DIN8 MKS70 Channel pressure 33 Channel 1 [d0 21]
05:30:46.812: Out 2- DIN8 MKS70 Channel pressure 31 Channel 1 [d0 1f]
05:30:46.819: Out 2- DIN8 MKS70 Channel pressure 29 Channel 1 [d0 1d]
05:30:46.826: Out 2- DIN8 MKS70 Channel pressure 27 Channel 1 [d0 1b]
05:30:46.833: Out 2- DIN8 MKS70 Channel pressure 25 Channel 1 [d0 19]
05:30:46.840: Out 2- DIN8 MKS70 Channel pressure 23 Channel 1 [d0 17]
05:30:46.847: Out 2- DIN8 MKS70 Channel pressure 21 Channel 1 [d0 15]
05:30:46.854: Out 2- DIN8 MKS70 Channel pressure 19 Channel 1 [d0 13]
05:30:46.861: Out 2- DIN8 MKS70 Channel pressure 17 Channel 1 [d0 11]
05:30:46.868: Out 2- DIN8 MKS70 Channel pressure 15 Channel 1 [d0 0f]
05:30:46.875: Out 2- DIN8 MKS70 Channel pressure 13 Channel 1 [d0 0d]
05:30:46.882: Out 2- DIN8 MKS70 Channel pressure 11 Channel 1 [d0 0b]
05:30:46.889: Out 2- DIN8 MKS70 Channel pressure 13 Channel 1 [d0 0d]
05:30:46.896: Out 2- DIN8 MKS70 Channel pressure 15 Channel 1 [d0 0f]
05:30:46.903: Out 2- DIN8 MKS70 Channel pressure 17 Channel 1 [d0 11]
05:30:46.910: Out 2- DIN8 MKS70 Channel pressure 19 Channel 1 [d0 13]
```
this is coming from my seaboard, which is the controller keyboard.
2. checkmarks in the menu besides the 2 options (✔) might be nice, so one knows what is currently active

**Comment by christofmuc** [comment]
Uh, wait, this is MIDI output, not input. Do you use the masterkeyboard feature and have the Orm reroute MIDI? It could be that I put the filter only on the input and not output, and here it seems the computer is sending the messages to Out 2?

**Comment by christofmuc** [comment]
Fixed in 2.1.0

**Comment by markusschloesser** [comment]
sorry never answered here, yes I sometimes use the seabaord as masterkeyboard
